### PR TITLE
Create v4_to_v5_migration_guide.md

### PR DESCRIPTION
**Pull request by aboedo**
Adds migration guide from v4 to v5. 

The actual guide is copied over from the v5.0.0 release notes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ Sign up to [get started for free](https://app.revenuecat.com/signup).
 
 React Native Purchases is the client for the [RevenueCat](https://www.revenuecat.com/) subscription and purchase tracking system. It is an open source framework that provides a wrapper around `StoreKit`, `Google Play Billing` and the RevenueCat backend to make implementing in-app purchases in `React Native` easy.
 
-## Migrating from Purchases v3
-- See our [Migration guide](https://revenuecat-docs.netlify.app/documentation/revenuecat/v4_api_migration_guide)
```
Comment (tonidero) [inline README.md:23]: Should we keep this old section as well? (not sure how many people would still be using v3...)

File: v4_to_v5_migration_guide.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+For more information around configuration please take a look at the [Amazon Store section in our docs](https://docs.revenuecat.com/docs/amazon-platform-resources). The official [Amazon In-App Purchasing docs](https://developer.amazon.com/docs/in-app-purchasing/iap-overview.html) also contain very valuable information, specially around testing and best practices.
+
+
+#### Apple AdServices support
+New method for automatic collecting of attribution tokens on iOS and macOS using AdServices - `Purchases.enableAdServicesAttributionTokenCollection()`
+
+#### New types and cleaned up naming
+New types that wrap native types from Apple, Google and Amazon, and we cleaned up the naming of other types and methods for a more consistent experience.
```
Comment (tonidero) [inline v4_to_v5_migration_guide.md:16]: FFTI. Should we add what new types were added? There is a section below for renames, but not for additions...

**Review by tonidero — APPROVED** [review]
LGTM!

File: README.md (hunk 1 above)
Comment (aboedo) [reply README.md:23]: the V3 there is a terrible copy-paste bug, we don't actually have a migration guide from v3 of React-Native. That was just pointing to the V3 -> V4 migration guide on iOS

**Pull request merged**